contracts/service-registry/src/state.rs

```diff
@@ -36,16 +36,24 @@ pub struct Worker {
     pub service_name: String,
 }
 
-impl TryInto<Participant> for Worker {
+impl TryFrom<Worker> for Participant {
     type Error = ContractError;
 
-    fn try_into(self) -> Result<Participant, ContractError> {
-        match self.bonding_state {
-            BondingState::Bonded { amount } => Ok(Participant {
-                address: self.address,
+    fn try_from(worker: Worker) -> Result<Participant, ContractError> {
+        match worker.bonding_state {
+            BondingState::Bonded { amount } => Ok(Self {
+                address: worker.address,
                 weight: amount.try_into()?,
             }),
-            _ => Err(ContractError::InvalidBondingState(self.bonding_state)),
+            _ => Err(ContractError::InvalidBondingState(worker.bonding_state)),
+        }
+    }
+}
+
+impl From<Worker> for NonWeightedParticipant {
+    fn from(worker: Worker) -> Self {
```

Comment:
How can it be the case that only bonded workers can become a weighted participant but non-bonded workers can become a non-weighted participant?